src/reanimated2/background.ts

```diff
@@ -0,0 +1,62 @@
+'use strict';
+import NativeReanimatedModule from './NativeReanimated';
+import { shouldBeUseWeb } from './PlatformChecker';
+import type { WorkletFunction } from './commonTypes';
+import type { WorkletRuntime } from './runtimes';
+import {
+  makeShareableCloneOnUIRecursive,
+  makeShareableCloneRecursive,
+} from './shareables';
+
+const SHOULD_BE_USE_WEB = shouldBeUseWeb();
+
+export type BackgroundQueue = {
+  __hostObjectBackgroundQueue: never;
+};
+
+export function createBackgroundQueue(name: string) {
+  return NativeReanimatedModule.createBackgroundQueue(name);
+}
+
+// @ts-expect-error Check `runOnUI` overload.
+export function runOnBackgroundQueue<Args extends unknown[], ReturnValue>(
+  backgroundQueue: BackgroundQueue,
+  workletRuntime: WorkletRuntime,
+  worklet: (...args: Args) => ReturnValue
+): WorkletFunction<Args, ReturnValue>;
+/**
+ * Schedule a worklet to execute on the background queue.
+ */
+export function runOnBackgroundQueue<Args extends unknown[], ReturnValue>(
+  backgroundQueue: BackgroundQueue,
+  workletRuntime: WorkletRuntime,
+  worklet: WorkletFunction<Args, ReturnValue>
+): (...args: Args) => void {
+  'worklet';
+  if (__DEV__ && !SHOULD_BE_USE_WEB && worklet.__workletHash === undefined) {
+    throw new Error(
+      '[Reanimated] The function passed to `runOnBackgroundQueue` is not a worklet.'
+    );
+  }
+  if (_WORKLET) {
+    return (...args) =>
+      _scheduleOnBackgroundQueue(
+        backgroundQueue,
+        workletRuntime,
+        makeShareableCloneOnUIRecursive(() => {
+          'worklet';
+          worklet(...args);
+        })
+      );
+  } else {
+    return (...args) =>
+      NativeReanimatedModule.scheduleOnBackgroundQueue(
+        backgroundQueue,
+        workletRuntime,
+        makeShareableCloneRecursive(() => {
+          'worklet';
+          worklet(...args);
+        })
+      );
+  }
```

Comment:
We need to have two separate implementations here, one for RN runtime and the other for worklet runtimes (including the UI runtime) because of two reasons:
1. In the RN runtime, we inject the native code using a HostObject with native methods, but in the UI (worklet) runtime we inject native code as individual functions. We'll need to unify the injecting logic.
2. In the RN runtime, we need to use `makeShareableCloneRecursive`, as opposed to `makeShareableCloneOnUIRecursive` on the UI (worklet) runtime. We'll need to unify the cloning logic.

Reply:
Should we change the name (or provide an alias) for `makeShareableCloneOnUIRecursive` to `makeShareableCloneOnWorkletRuntime`?

Reply:
Yep, we could do it